File: pipeline/templates/estimation.r
suppressMessages(library(tidyverse))

{% include 'load_data.r' %}

df = suppressMessages(load_data("{{ depends_on }}"))

df %>% mutate_if(is.character, as.factor) -> df

{% block estimation_method %}{% endblock %}

saveRDS(model, "{{ produces }}")

File: pipeline/templates/estimation.py
import numpy as np
import joblib
import pandas as pd
import statsmodels.formula.api as smf


{% include 'load_data.py' %}


def fit_model(df):
    {% block estimation_method %}{% endblock %}
    return model


def save_model(model):
    joblib.dump(model, "{{ produces }}")


def main():
    df = load_data("{{ depends_on }}")
    model = fit_model(df)
    save_model(model)


if __name__ == "__main__":
    main()
